packages/iTwinUI-react/src/core/Header/Header.tsx

```diff
@@ -68,6 +71,20 @@ export type HeaderProps = {
    * </DropdownMenu>
    */
   userIcon?: React.ReactNode;
+  /**
+   * Avatar
+   *
+   * It's size and transition will be handled between slim/not slim state of the
+   * Header
+   * (expects `Avatar`, can be wrapped in `IconButton` and `DropdownMenu` if needed)
+   * @example
+   * <DropdownMenu menuItems={...}>
+   *   <IconButton styleType='borderless'>
+   *     <Avatar ... />
+   *   </IconButton>
+   * </DropdownMenu>
+   */
+  avatar?: React.ReactNode;
```

Comment:
1. I don't think changes to the `Header` API should be part of this PR
2. I don't think we need a new `avatar` prop. We can just deprecate `userIcon` and suggest using `actions` instead. See #367

Reply:
**[DONE]** Removed `avatar`.

I think `userIcon` might have been there to enforce the `UserIcon` to always be to the left of `menuItems`. Without it, it can be anywhere in the `actions`.

But even deprecating it makes sense. Since we have a `Header` stories to show where we recommend to place `Avatar` in `actions`. It is also alright since, like you said, there was nothing special being done specifically to the `userIcon`.

Reply:
Just to be clear: you need to bring back `userIcon`. Deprecating does not mean removing it (you have removed it from the header code).

Reply:
Ohh, so deprecate means it _should work even if users use it_, but we do not recommend it as we won't maintain it and it also might be removed in the future?

So deprecate doesn't mean that it won't work anymore so you _must_ change to what we recommend?

Reply:
> Just to be clear: you need to bring back `userIcon`.

Brought `userIcon` back to `Header`. Also added some JSDocs to `actions` to recommend users to add `Avatar` to the end of the `actions` array ([L43-46](https://github.com/iTwin/iTwinUI-react/pull/902/commits/4120bb19e67d3b48175032a715d4bdfd064bbf42#diff-c792e3ac6e0ca0e4361d7efffdd239f466a22831e373546c4e9fd52c786e37ebR43-R46)).